grabbit/src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -0,0 +1,56 @@
+package com.twcable.grabbit.jcr
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.apache.jackrabbit.value.DateValue
+
+import javax.annotation.Nonnull
+import javax.jcr.Node as JCRNode
+import javax.jcr.RepositoryException
+
+import static org.apache.jackrabbit.JcrConstants.JCR_LASTMODIFIED
+import static org.apache.jackrabbit.JcrConstants.JCR_PRIMARYTYPE
+
+@CompileStatic
+@Slf4j
+class JCRNodeDecorator {
+
+    @Delegate
+    JCRNode innerNode
+
+
+    JCRNodeDecorator(@Nonnull JCRNode node) {
+        if(!node) throw new IllegalArgumentException("node must not be null!")
```

Comment:
You are throwing the exception here but where it has been handled? In case of exception what will happen to the flow?